### DISCUSSION TITLE
command / control API?

### DISCUSSION THREAD

**Discussion by Syd-Ra** [nasa/openmct#3854] [Community Q&A] [answered]
Hi Everyone, we're considering building a satellite mission control solution using openMCT. The telemetry monitoring capabilities are great; however I was not able to find much discussion on the ability to issue commands. I did see the Yamcs plugin, but given that Yamcs is a whole other monitoring/command application and the "MCT" stands for "mission control technologies" I was hoping there might be a builtin solution. 

Commanding could be via a sequence of text entry, confirm, send, acknowledge, or graphical by clicking a UI element. Does this capability (or a predefined extension point for it) currently exist? 

Thanks!

**Comment by akhenry** [comment, marked as answer]
@Syd-Ra Hi! Open MCT doesn't have a native commanding interface. You mentioned YAMCS, which Open MCT works really well with, and which does support commanding. If you were to use YAMCS as the backend, you could in principle embed YAMCS commanding pages into Open MCT displays using our Web Page object type. 

Getting started with YAMCS is pretty straightforward using their [Quick Start repo](https://github.com/yamcs/quickstart), and [the Open MCT-YAMCS project](https://github.com/akhenry/openmct-yamcs) is designed to work out of the box with YAMCS Quick Start, so it might be worth pulling them both down and just giving it a go to see if that combination works for you.

**Reply by Syd-Ra** [reply]
Thanks the reply - I'll read up on YAMCS some more and see where it goes.
